### PR TITLE
(SIMP-8081) Upgrade Instructions for SIMP 6.5.0

### PR DESCRIPTION
**Pull request by jeannegreulich**

### CODE REVIEW

File: docs/user_guide/Upgrade_SIMP/General_Upgrade_Instructions.rst [hunk 1]
```diff
@@ -108,9 +108,16 @@ steps as ``root``:
    ** **This ends the steps that are only for 6.4 or later.**  The next steps apply
    to all systems.
 
+#. Update the generated types for the environment
+
+   .. code-block:: sh
+
+     puppet generate types  --environment test
```
Comment (op-ct) [inline docs/user_guide/Upgrade_SIMP/General_Upgrade_Instructions.rst:115]:
```suggestion
     puppet generate types  --environment production
```

Comment (trevor-vaughan) [reply docs/user_guide/Upgrade_SIMP/General_Upgrade_Instructions.rst:115]: Did you need to set your `umask` for this? I don't quite remember.

If we have them fire off `simp_generate_types` on all spaces it should "do the right thing"

**Comment by op-ct** [comment]
Rebased + updated with various content + formatting fixes, including changelog + sub-TOCs

**Review by op-ct — APPROVED** [review]
:eye: :+1:

**Review by op-ct — DISMISSED** [review]
:eye: 👍 #️⃣ :two:

**Review by op-ct — APPROVED** [review]
:eye: 👍 #️⃣ 3️⃣

**Comment by op-ct** [comment]
Travis CI checks failing linkcheck due to apparently localized SSL errors; Merged

**Pull request merged**